core/tests/performance_framework/process_data.py

```diff
@@ -0,0 +1,117 @@
+
+We need selenium + browsermob-proxy to capture page loading session
+(or HTTP Archive), to obtain page size stats. However, owing to the sluggish
+nature of the proxy server, timing stats are not accurate.
+Hence, we have separate timing stats which are obtained without making
+use of the proxy, directly from the browser console.
+"""
+
+
+class PageSessionMetricsRetriever(object):
+    """Contains methods to process stats and provide performance metrics.
+
+    page_session_stats if a dictionary containing page load statistics or
+    HTTP Archive (HAR).
+    Refer: https://dvcs.w3.org/hg/webperf/raw-file/tip/specs/HAR/Overview.html
+
+    page_session_timings is a dictionary containing various timings associated
+    with a page load like request time, dom load time, etc.
+    Structure:
+        {
+          "webkitClearResourceTimings": {},
+          "memory": {...},
+          "timing": {
+            "domLoading": 1465470586738,
+            "fetchStart": 1465470583251,
+          }
+        }
+    """
+
+    def __init__(self, page_session_stats, page_session_timings):
+        self.page_session_stats = page_session_stats
+        self.page_session_timings = page_session_timings
+
+        self.page_load_timings = {}
+        if 'timing' not in self.page_session_timings:
+            raise ValueError('page_session_timings missing key: timing')
+
+        self.page_load_timings = page_session_timings['timing']
+
+    def get_request_count(self):
+        """Return no of requests made for the page load."""
+        request_count = 0
+
+        if 'log' in self.page_session_stats:
+            if 'entries' in self.page_session_stats['log']:
+                request_count = len(self.page_session_stats['log']['entries'])
+
+        return request_count
+
+    def get_total_page_size_bytes(self):
+        """Return total page size which includes the size of all resources
+        for a page.
+        """
+        total_size = 0
+
+        for entry in self.page_session_stats['log']['entries']:
+            total_size += int(entry['response']['bodySize'])
+
+        print 'total_size: ', total_size
+
+        if total_size < 0:
+            raise Exception('Total page size cannot be negative.')
+
+        return total_size
+
+    def _get_duration_secs(self, event_end, event_initial):
+        # Check if timestamps are seconds or milliseconds.
+        # From: http://goo.gl/iHNYWx
+        if event_initial not in self.page_load_timings:
+            raise ValueError('page_load_timings missing key: %s' %event_initial)
+
+        initial_timestamp = self.page_load_timings[event_initial]
+
+        if event_end not in self.page_load_timings:
+            raise ValueError('page_load_timings missing key: %s' %event_end)
+
+        end_timestamp = self.page_load_timings[event_end]
+
+        # If milliseconds convert to seconds.
+        if len(str(initial_timestamp)) >= 13:
```

Comment:
If it is definitely going to be in milliseconds, we probably should keep it in milliseconds for consistency. Also, like @seanlip mentioned, we shouldn't need to do a seconds/milliseconds conversion check.